core/server/api/canary/users.js

```diff
@@ -230,5 +233,31 @@ module.exports = {
                 return models.ApiKey.refreshSecret(model.toJSON(), Object.assign({}, {id: model.id}));
             });
         }
+    },
+
+    resetAllPasswords: {
+        headers: {
+            cacheInvalidate: true
+        },
+        permissions: async (frame) => {
+            // TODO: Is there an easier way than querying the roles?
+            const user = await models.User.findOne({
```

Comment:
Is there a related resource which has similar permission level needed? We might need to add entries to permissions table instead :thinking:  Definitely should not be doing checks like this here and leave it for the permission layer to handle  :wink: